security/pfSense-pkg-snort/files/usr/local/www/snort/snort_barnyard.php

```diff
@@ -178,7 +178,14 @@
 		if ($_POST['barnyard_dbhost']) $natent['barnyard_dbhost'] = $_POST['barnyard_dbhost']; else unset($natent['barnyard_dbhost']);
 		if ($_POST['barnyard_dbname']) $natent['barnyard_dbname'] = $_POST['barnyard_dbname']; else unset($natent['barnyard_dbname']);
 		if ($_POST['barnyard_dbuser']) $natent['barnyard_dbuser'] = $_POST['barnyard_dbuser']; else unset($natent['barnyard_dbuser']);
-		if ($_POST['barnyard_dbpwd']) $natent['barnyard_dbpwd'] = base64_encode($_POST['barnyard_dbpwd']); else unset($natent['barnyard_dbpwd']);
+
+		// The password field will return '********' if no changes are made and needs to be escaped.
+        if ($_POST['barnyard_dbpwd'] && ($_POST['barnyard_dbpwd'] != DMYPWD)) $natent['barnyard_dbpwd'] = base64_encode($_POST['barnyard_dbpwd']);
+        else
+			// Because of the base64 encoding/decoding, in the case of a valid value that hasn't changed, it needs to be re-encoded to base64.
+            if ($_POST['barnyard_dbpwd'] != DMYPWD) unset($natent['barnyard_dbpwd']);
+            else $natent['barnyard_dbpwd'] = base64_encode($natent['barnyard_dbpwd']);
```

Comment:
This change is essentially identical to the one submitted for Suricata.  The same issue with re-encoding an already encoded password upon a "save" operation exists in Snort since both packages share much of the same GUI code.  I am OK with this fix, but I do agree the if/then/else statement indents need fixing up.

Bill